public/components/trace_analytics/components/common/helper_functions.tsx

```diff
@@ -615,3 +616,77 @@ export const generateServiceUrl = (
 
   return url;
 };
+
+/*
+ * Parse an ISO timestamp with up to nanosecond precision.
+ * For example, "2025-01-28T03:12:37.293990144Z" will be converted
+ * to a number representing the total nanoseconds since the Unix epoch.
+ */
+export function parseIsoToNano(iso: string): number {
+  const match = iso.match(/^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2})(?:\.(\d+))?Z$/);
+  if (!match) {
+    throw new Error(`Invalid ISO timestamp: ${iso}`);
+  }
+  // Parse the base part using Date.parse (which gives ms)
+  const baseMs = new Date(match[1] + 'Z').getTime();
+  // Get the fractional part (if any), pad to 9 digits for nanosecond precision
+  let fraction = match[2] || '0';
+  fraction = fraction.padEnd(9, '0'); // ensure it has 9 digits
+  return baseMs * NANOS_TO_MS + Number(fraction);
+}
+interface Span {
+  traceId: string;
+  spanId: string;
+  traceState: string;
+  parentSpanId: string;
+  name: string;
+  kind: string;
+  startTime: string;
+  endTime: string;
+  durationInNanos: number;
+  serviceName: string;
+  events: any[];
+  links: any[];
+  droppedAttributesCount: number;
+  droppedEventsCount: number;
+  droppedLinksCount: number;
+  traceGroup: string;
+  traceGroupFields: {
+    endTime: string;
+    durationInNanos: number;
+    statusCode: number;
+  };
+  status: {
+    code: number;
+  };
+  instrumentationLibrary: {
+    name: string;
+    version: string;
+  };
+}
+
+interface ParsedHit {
```

Comment:
Can we please move these to common constants?

Reply:
Moved to constants